src/util/cache.ts

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+export class Cache extends Map {
+  private static _instance: Cache;
+  private _hits = 0;
+  private _lookups = 0;
+  private _enabled = true;
+
+  public static instance(): Cache {
+    if (!Cache._instance) {
+      Cache._instance = new Cache();
+    }
+    return Cache._instance;
+  }
+
+  public static set<V>(key: string, value: V) {
```

Comment:
what's the `V` for?

cache.set<string>('key', 4) // compile error?

Reply:
@mshanemc `V` is type of the value. So in your example, `cache.set('key', 4)` will compile properly. I believe TS will infer the type from the type of the value provided in the set call.

Reply:
sorry, type-o.  I meant to ask about the `<T>` outside of the method, saying that

`cache.set<'string'>('key', 4)` would fail.  Trying to figure out what the purpose of that is